### PR TITLE
Fix unused variable warnings

### PR DESCRIPTION
**Pull request by jenningsje**
I'll be checking this over to see if it correct or if there are bugs within the code.

### CODE REVIEW

File: src/sos/section_analyzer.py [hunk 1]
```diff
@@ -398,7 +398,7 @@ def get_step_input(section, default_input):
                 step_input = sos_targets()
             else:
                 step_input = default_input
-        elif not any(isinstance(x, (dynamic, remote)) for x in args):
+        elif not any(isinstance(x, (dynamic, remote)) for x in kwargs):
```
Comment (BoPeng) [inline src/sos/section_analyzer.py:401]: You changed the logic of the code, which is wrong. I do not know why there is a warning here but `args` was indeed defined in line 394.

File: src/sos/step_executor.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 import os
 import subprocess
 import sys
+from textwrap import dedent
```
Comment (BoPeng) [inline src/sos/step_executor.py:11]: Why is this? dedent is not used in this module.

File: src/sos/task_executor.py [hunk 3]
```diff
@@ -401,7 +401,7 @@ def execute_master_task(self, task_id, params, master_runtime, sig_content):
         # node. The "n_procs" is the "number of cpus per job per node", which,
         # is "cores" (per task) * "n_workers" (per node). Therefore,
         # "n_procs" should not be used.
-        n_nodes, n_procs = self._parse_num_workers(env.config["worker_procs"])
+        n_nodes = self._parse_num_workers(env.config["worker_procs"])
```
Comment (BoPeng) [inline src/sos/task_executor.py:404]: change to `n_nodes, _`, or betteer `n_nodes = .... [0]`.

File: src/sos/workflow_executor.py [hunk 4]
```diff
@@ -1635,7 +1635,7 @@ def run_as_master(self, targets=None, mode=None) -> Dict[str, Any]:
         except KeyboardInterrupt as e:
             if exec_error.errors:
                 failed_steps, pending_steps = dag.pending()
-                if pending_steps:
+                if failed_steps:
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1638]: This changes the logic of the code. If `failed_steps` is not used, do `pending_steps = dag.pending()[1]`.

File: src/sos/workflow_executor.py [hunk 5]
```diff
@@ -1907,15 +1907,15 @@ def run_as_nested(self,
...
                     sections = [
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1911]: ditto

File: src/sos/workflow_executor.py [hunk 6]
```diff
@@ -1907,15 +1907,15 @@ def run_as_nested(self,
...
                         self.workflow.section_by_id(
                             x._step_uuid).step_name(True) for x in pending_steps
                     ]
                     exec_error.append(
                         self.workflow.name,
                         RuntimeError(
-                            f'{len(sections)} pending step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
+                            f'{len(sections)} failed step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1918]: please do not change the logic of the code.

File: src/sos/workflow_executor.py [hunk 7]
```diff
@@ -1937,15 +1937,15 @@ def run_as_nested(self,
             # exec_error.append(self.workflow.name,
             #    RuntimeError('{} failed step{}: {}'.format(len(sections),
             #        's' if len(sections) > 1 else '', ', '.join(sections))))
-            if pending_steps:
+            if failed_steps:
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1940]: ditto

File: src/sos/workflow_executor.py [hunk 8]
```diff
@@ -1662,7 +1662,7 @@ def run_as_master(self, targets=None, mode=None) -> Dict[str, Any]:
         )
 
         if exec_error.errors:
-            failed_steps, pending_steps = dag.pending()
+            pending_steps = dag.pending()
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1665]: this is still wrong.

File: src/sos/workflow_executor.py [hunk 9]
```diff
@@ -1945,7 +1945,7 @@ def run_as_nested(self,
                 exec_error.append(
                     self.workflow.name,
                     RuntimeError(
-                        f'{len(sections)} pending step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
+                        f'{len(sections)} failed step{"s" if len(sections) > 1 else ""}: {", ".join(sections)}'
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1948]: this should not have changed.

File: src/sos/workflow_executor.py (hunk 8 above)
Comment (BoPeng) [reply src/sos/workflow_executor.py:1665]: @jenningsje Please fix this PR so that I can merge.

File: src/sos/task_executor.py [hunk 10]
```diff
@@ -401,7 +401,7 @@ def execute_master_task(self, task_id, params, master_runtime, sig_content):
         # node. The "n_procs" is the "number of cpus per job per node", which,
         # is "cores" (per task) * "n_workers" (per node). Therefore,
         # "n_procs" should not be used.
-        n_nodes, n_procs = self._parse_num_workers(env.config["worker_procs"])
+        n_nodes, _ = self._parse_num_workers(env.config["worker_procs"])[0]
```
Comment (BoPeng) [inline src/sos/task_executor.py:404]: What are you doing here?

Comment (BoPeng) [reply src/sos/task_executor.py:404]: _ is not needed if you add [0].

Comment (jenningsje) [reply src/sos/task_executor.py:404]: I just changed n_nodes, _   self._parse_num_workers(env.config["worker_procs"])[0] to n_nodes, _   self._parse_num_workers(env.config["worker_procs"]) and I pushed the changes but I do not see them on GitHub.

Comment (BoPeng) [reply src/sos/task_executor.py:404]: check if you are on the right branch.

**Review by BoPeng — APPROVED** [review]

**Comment by BoPeng** [comment]
tests fail.

File: src/sos/workflow_executor.py [hunk 11]
```diff
@@ -1634,7 +1634,7 @@ def run_as_master(self, targets=None, mode=None) -> Dict[str, Any]:
                     time.sleep(0.1)
         except KeyboardInterrupt as e:
             if exec_error.errors:
-                failed_steps, pending_steps = dag.pending()
+                failed_steps, pending_steps = dag.pending()[1]
```
Comment (BoPeng) [inline src/sos/workflow_executor.py:1637]: This is not right....

Comment (jenningsje) [reply src/sos/workflow_executor.py:1637]: Sorry. I'll look back at your previous comment for this line.

**Comment by BoPeng** [comment]
Numerous commits and discussions for 9 actual changes....

**Pull request merged**